### ISSUE TITLE
 No detect audio jack Intel Corporation Celeron/Pentium Silver Processor High Definition Audio (rev 06)

### ISSUE THREAD

**Issue by jirisarri10** [thesofproject/linux#3460]
Hi there,
i have got Linux 5.16 and not detects headphones! however usb port detect headphones and work!
Thank you!

**Comment by plbossart** [comment]
please attach the result of 'alsa-info' @jirisarri10, you did not provide any information that we can use.

**Comment by IgorNovozhilov** [comment]
@plbossart
Hi, I have a similar problem. May my information help to fixed the problem.
system info:
```
System:    Kernel: 5.13.0-30-generic x86_64 bits: 64 compiler: N/A Desktop: Cinnamon 5.2.7 
           wm: muffin dm: LightDM Distro: Linux Mint 20.3 Una base: Ubuntu 20.04 focal 
Machine:   Type: Desktop System: Chatreey product: AC1-DP v: N/A serial: <filter> 
           Mobo: N/A model: N/A serial: <filter> UEFI: American Megatrends 
           v: GB08_SVID_CHATREEY_0_01 date: 10/29/2021 
Audio:     Device-1: Intel driver: sof-audio-pci-intel-apl bus ID: 00:0e.0 chip ID: 8086:3198 
           Sound Server: ALSA v: k5.13.0-30-generic 
```
```
$ lspci -nnk | grep -i audio -A2
00:0e.0 Audio device [0403]: Intel Corporation Device [8086:3198] (rev 06)
	DeviceName: Onboard - Sound
	Subsystem: Device [0301:02f3]
	Kernel driver in use: sof-audio-pci-intel-apl
	Kernel modules: snd_hda_intel, snd_soc_skl, snd_sof_pci_intel_apl
00:0f.0 Communication controller [0780]: Intel Corporation Celeron/Pentium Silver Processor Trusted Execution Engine Interface [8086:319a] (rev 06)
```

[alsa-info.txt](https://github.com/thesofproject/linux/files/8207971/alsa-info.txt)

**Comment by plbossart** [comment]
@IgorNovozhilov you have an ES8336 codec, you will need to use PR #3338 and likely additional patches to support this device.
````
/sys/bus/acpi/devices/ESSX8336:00/status 	 15
````
There's no way your 5.13 kernel will work on such a device.

**Comment by IgorNovozhilov** [comment]
@plbossart In which kernel release can we expect these fixes?

**Comment by plbossart** [comment]
> @plbossart In which kernel release can we expect these fixes?

5.18 at best, I just sent the patches to the maintainers.

**Comment by jirisarri10** [comment]
Thank you very much, Pierre-Louis.
In the end I couldn't send any info because I installed Windows 10 (God help me!!!). I understand that in the new kernel 5.17 ES833 will still not work. A hug!!

**Comment by nikeedev** [comment]
> > @plbossart In which kernel release can we expect these fixes?
> 
> 5.18 at best, I just sent the patches to the maintainers.

Maybe. 
A little bit about why im here:
I have a laptop that because of that Linux and intel issue, linux doesnt recognize the pc speakers. I use Manjaro as distro, but the speakers didn't work even at other distros. 

Now back to issue.

I have tested speakers in 5.19 experimental, and it did recognize the speakers but didn't produce any sound, so that means that they are working on. I have also tested on 5.18 which didn't recognize as in beginning. 

This means that on most devices it won't be available till 5.19 or even 5.20, and probably not on 5.18 as you thought....I will wait till that time comes.
